lib/_stream_transform.js

```diff
@@ -162,7 +162,7 @@ Transform.prototype._write = function(chunk, encoding, cb) {
 Transform.prototype._read = function(n) {
   var ts = this._transformState;
 
-  if (ts.writechunk !== null && ts.writecb && !ts.transforming) {
```

Comment:
> this change also means that writing null to a passthrough will end the readable side

I don't think so.  [`null` and `undefined` are ignored by the `._transform` callback.](https://github.com/nodejs/node/blob/v5.9.0/lib/_stream_transform.js#L76)